garbage-collect/handler.go

```diff
@@ -48,6 +55,28 @@ func Handle(req []byte) string {
 	return fmt.Sprintf("Garbage collection ran for %s/%s - %d functions deleted.", garbageReq.Owner, garbageReq.Repo, deleted)
 }
 
+func validateRequest(req []byte) (err error) {
+	payloadSecret, err := sdk.ReadSecret("payload-secret")
+
+	if err != nil {
+		return fmt.Errorf("couldn't get payload-secret: %t", err)
+	}
+
+	xCloudSignature := os.Getenv("Http_X_Cloud_Signature")
+
+	if xCloudSignature == "" {
+		return fmt.Errorf("unauthorized request")
+	}
+
+	err = hmac.Validate(req, xCloudSignature, payloadSecret)
+
+	if err != nil {
+		return fmt.Errorf("unauthorized request")
```

Comment:
What error message are we using elsewhere when the digest doesn't match or is invalid? We should aim to be consistent across the board to help with maintenance.